.github/workflows/release.yml

```diff
@@ -4,17 +4,24 @@
 name: Release
 on:
   workflow_dispatch:
+  schedule:
+    # 9:25 UTC every 1th and 15th of month
+    - cron: "25 09 1,15 * *"
```

Comment:
This can be set to run every three weeks (at 9:25am) like this: `25 9 */21 * *` . I think you can also set the day of the week you want the release by setting the last value. So, for example `25 9 */21 * 5` would be every three weeks on Friday at 9:25am.

See the [CronJob docs](https://docs.openshift.com/container-platform/3.11/dev_guide/cron_jobs.html) for more information.

Reply:
I think what you proposed just execute the 21th day of each month, so once per month. https://crontab.guru/#*_*_*/21_*_*

Reply:
Based on [crontab guru](https://crontab.guru/#25_9_*/21_*_*) Chris suggestion runs each 21 days starting on 1st of the month, meaning, runs twice a month on day 1 and 22, but that's not what's intended.

Ideally I think, that it will be nice to have this run a specific weekday to ensure is not a saturday for example, my proposal isn't as plain as `25 09 1,15 * *` but maybe we want to use it.
It's the combination of this crontab: `25 09 1-7,15-21 * *` plus, checking if the day is a specific day of the week (e.g, Tuesday):

```yaml
on:
  schedule:
    - cron: '25 09 1-7,15-21 * *' # Run every day at 09:25 each the first and third week each month.

env: 
  RELEASE_DAY_OF_WEEK: 2 # Same numbers used in cron

jobs:
  setup:
    uses: ./.github/workflows/log_run_details.yml
  set-day-of-week:
    outputs:
      - dayOfWeek: ${{ steps.setDayOfWeek.outputs.day }}
    steps:
      - id: setDayOfWeek
        run: |
          DOW=$(date +%u)
          echo "dow=${DOW}" >> "$GITHUB_OUTPUT"
  release:
    needs: [set-day-of-week]
    # Run manually or (scheduled and upstream repository), so don't allow for scheduled on forks
    if: github.event_name == "workflow_dispatch" || (github.event_name == "schedule" && github.repository == 'kubearchive/kubearchive') && (needs.set-day-of-week.outputs.dayOfWeek == ${RELEASE_DAY_OF_WEEK})
    steps:
     ...
```

The code is untested but I followed the following references:
* https://github.com/orgs/community/discussions/25941
* https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#defining-environment-variables-for-a-single-workflow

Reply:
I don't think its a big deal if we do a release on a weekend. We can review how it went when we get back into weekdays.